src/develop/masks/masks.c

```diff
@@ -35,6 +36,144 @@
 #include "develop/masks/group.c"
 // clang-format on
 
+typedef struct _masks_undo_data_t
+{
+  GList *forms;
+  dt_masks_form_t *form;
+} _masks_undo_data_t;
+
+static void _masks_write_form_db(dt_masks_form_t *form, dt_develop_t *dev);
+// write a form into the database
+
+static void _masks_write_forms_db(dt_develop_t *dev, gboolean undo);
+// write all masks form into the database. record an undo is undo is true
+
+static void _masks_free_form_list(GList *forms)
```

Comment:
Why this function? Isn't that the same as `g_list_free_full(forms, dt_masks_free_form);`?